File: PCD/Atividades/Atv1/Java/rleSequencial/src/rleSequencial.java
package PCD.Atividades.Atv1.Java.rleSequencial.src;
public class rleSequencial {
    public static String compressRle(String input){
        String output = "";
        char aux = input.charAt(0);
        String joinAux;
        int count = 1;

        for(int i = 1; i < input.length(); i++){
            if(input.charAt(i) == aux){         //if the next is the same char, count+1
                count++;
            }
            else{                   //else, save the count of chars and add in the output string
                joinAux = String.valueOf(aux)+String.valueOf(count);
                output = output.concat(joinAux);
                aux = input.charAt(i);
                count = 1;
            }
        }
        return output;
    }

      public static void main(String[] args){
        String input = "aaabbccccddddeeeeeffaaaaaaa";
        String output;
        long elapsedTime;

        long startTime = System.nanoTime();
        output = compressRle(input);
        long endTime = System.nanoTime();
        elapsedTime = endTime - startTime;


        System.out.println("String comprimida: " + output);
        System.out.println("Tempo de execução (sequencial): " + (double) elapsedTime / 1_000_000_000);

    }
}

File: PCD/Atividades/Atv1/C/rleSequencial.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

void compressRle(const char* input, char* output) {
    int count, i = 0, j = 0;
    while (input[i] != '\0') {
        output[j++] = input[i];
        count = 1;
        while (input[i] == input[i + 1]) {
            count++;
            i++;
        }
        j += sprintf(&output[j], "%d", count);
        i++;
    }
    output[j] = '\0';
}

int main() {
    char input[] = "aaabbccccddddeeeeeffaaaaaaa";
    char output[100];

    clock_t start = clock();
    compressRle(input, output);
    clock_t end = clock();

    double time_spent = (double)(end - start) / CLOCKS_PER_SEC;
    printf("String comprimida: %s\n", output);
    printf("Tempo de execução (sequencial): %f segundos\n", time_spent);

    return 0;
}
